.github/api_best_practices.md

```diff
@@ -0,0 +1,229 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [Service Name](#service-name)
+  - [RPC Method Name](#rpc-method-name)
+  - [RPC Message Name](#rpc-message-name)
+
+## Naming Guidelines
+
+### Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All RPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+All service defined in versioned package [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/v2)
+**must not** import from [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) and
+[/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) packages.
+All new RPC methods defined in versioned package [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/v1) **must not** import from
+`/proto/storage` and `/proto/internalapi` packages.
+
+### RPC Method Name
```

Comment:
(*)

Caring about less experienced readers, I suggest to stick with `gRPC` instead of "generic" `RPC` which can create confusion.

```suggestion
### gRPC Method Name
```

Please also search-replace `RPC` with `gRPC` in the entire doc.

Reply:
fair enough